src/main/java/org/aksw/simba/lemming/creation/Inferer.java

```diff
@@ -290,93 +309,140 @@ public OntModel readOntology(String filePath, String base) {
-		Iterator<Entry<String, Equivalent>> it = classes.entrySet().iterator();
+	public void renameClasses(Model model, Map<OntClass, String> classes) {
+		Iterator<Entry<OntClass, String>> it = classes.entrySet().iterator();
 		while (it.hasNext()) {
-			Map.Entry<String, Equivalent> pair = it.next();
-			String newName = pair.getValue().getName();
-			Resource mResource = model.getResource(pair.getKey());
-			if (mResource != null && !mResource.getURI().equals(newName)) {
-				ResourceUtils.renameResource(mResource, newName);
+			Map.Entry<OntClass, String> pair = it.next();
+			Resource mResource = model.getResource(pair.getKey().getURI());
+			if (mResource != null && !mResource.getURI().equals(pair.getValue())) {
+				ResourceUtils.renameResource(mResource, pair.getValue());
+			}
+		}
+	}
+
+	/**
+	 * Find a proper name to represent each equivalent resources set from the given map.
+	 * @param iriEquiMap a map that maps a resource to its equivalent resources set.
+	 * @return a map that maps a resource to a string name which represents the its corresponding equivalent resources set.
+	 */
+	private <T extends OntResource> Map<T, String> findProperName(Map<T, Set<T>> iriEquiMap){
+		Map<T, String> iriToName = new HashMap<>();
+		for(T re : iriEquiMap.keySet()){
+			if(!iriToName.containsKey(re)){
+				String properName = re.getURI();
+				Set<T> equis = iriEquiMap.get(re);
+				for(T eq : equis){
+					if(properName.compareTo(eq.getURI())<0){
+						properName = eq.getURI();
+					}
+				}
+				for(T eq : iriEquiMap.get(re)){
+					iriToName.put(eq, properName);
+				}
+			}
+		}
+		return  iriToName;
+	}
+
+	/**
+	 * Add domains and ranges of all equivalent ontproperties to the an ontproperty which represents all equivalent ontproperties.
+	 * @param propertyEquiMap map each ontproperty to a set of properties which contains its equivalent ontproperties.
+	 * @return a map that maps each ontproperty of given map to itself which carries all domains and ranges of its equivalent resources.
+	 */
+	private Map<OntProperty, OntProperty> collectRangeDomain(Map<OntProperty, Set<OntProperty>> propertyEquiMap) {
+		Map<OntProperty, OntProperty> ontProToOntProDR = new HashMap<>();
+		for (OntProperty property : propertyEquiMap.keySet()) {
+			OntProperty collectedProperty = property;
+			for (OntProperty currentProperty : propertyEquiMap.get(property)) {
+				List<? extends OntResource> domainList = currentProperty.listDomain().toList();
+				for (OntResource domain : domainList) {
+					if (!collectedProperty.hasDomain(domain)) {
+						collectedProperty.addDomain(domain);
+					}
+				}
+
+				List<? extends OntResource> rangeList = currentProperty.listRange().toList();
+				for (OntResource range : rangeList) {
+					if (!collectedProperty.hasRange(range)) {
+						collectedProperty.addRange(range);
+					}
+				}
 			}
+			ontProToOntProDR.put(property, collectedProperty);
 		}
+		return ontProToOntProDR;

```

Comment:
> The project is not building correctly at the moment, as there are compilation errors in `PrecomputingValuesTest`.
> 
> I think a unit test case for the equivalent resources search would be nice to have. @MichaelRoeder What do you think?

@aalexandrasilva Yes, the PrecomputingValuesTest is failed, because it only tests whether the generated models with the new Inferer are the same as the old inferer. Now they are not same, because the order of iterateStmts and renameClasses in Inferer are exchanged. I think, maybe the PrecomputingValueTest could be removed.

Reply:
From what I see it comes from a compilation error because `Equivalent` class has been deleted.

Reply:
From what I see the error comes from a compilation error since Equivalent class has been deleted.
However, I do agree with removing PrecomputingValueTest and have new unit test cases instead to check the functionality. But we should wait on @MichaelRoeder's opinion on this.

Reply:
I fully agree with @aalexandrasilva. It does not make much sense to keep the PrecomputingValueTest if it solely relies on the comparison of the old and the new Inferer implementations. It makes much more sense to implement tests that focus on the functionality that we want to achieve.